src/app/componenets/launch-page/launch-page.component.ts

```diff
@@ -0,0 +1,207 @@
+import { ILaunches } from 'src/app/interfaces/launches';
+
+@Component({
+  selector: 'app-launch-page',
+  templateUrl: './launch-page.component.html',
+  styleUrls: ['./launch-page.component.scss'],
+  providers: [
+    {
+      provide: STEPPER_GLOBAL_OPTIONS,
+      useValue: { displayDefaultIndicatorType: false },
+    },
+  ],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class LaunchPageComponent implements OnInit {
+
+  @ViewChild('stepperGround') private stepperGround: MatStepper;
+  @ViewChild('stepperAir') private stepperAir: MatStepper;
+
+  public groundProgressbarValue: BehaviorSubject<number> = new BehaviorSubject(100);
+  public airProgressbarValue: BehaviorSubject<number> = new BehaviorSubject(100);
+
+  public secTillLiftoff: BehaviorSubject<number> = new BehaviorSubject(0);
+  public secInAir: BehaviorSubject<number> = new BehaviorSubject(0);
+
+  public launch: ILaunches;
+
+  public containImages: boolean = false;
+  public images: Array<any> = [];
+  public widthPX: number = 0;
+
+  public containTimeLine: boolean = false;
+  public groundTimeline: ITimeline[] = new Array();
+  public airTimeline: ITimeline[] = new Array();
+
+  // Offset as number 1
+  private offset: number = 1;
+  private timeInterval: number = 100;
+
+  constructor(
+    public spinnerService: SpinnerService,
+    private route: ActivatedRoute,
+    private httpClient: HttpClient,
+    private sanitizer: DomSanitizer,
+    private location: Location,
+    private cdref: ChangeDetectorRef,
+  ) {}
+
+  @HostListener('window:resize', ['$event'])
+  onResize() {
+    this.widthPX = window.innerWidth;
+  }
+
+  public goBack(): void {
+    this.location.back();
+  }
+
+  public renderYT(YTid: any): SafeResourceUrl {
+    return this.sanitizer.bypassSecurityTrustResourceUrl(
+      'https://www.youtube.com/embed/' + YTid + '?autoplay=1&mute=1'
+    );
+  }
+
+  public imagesToArray(images: []): void {
+    this.images = images.map((image) => ({
+      name: image,
+    }));
+  }
+
+  public timelineStepperSetter(
+    timeline: ITimeline[],
+    startFrom: number,
+    currentSeconds: number,
+    stepper: MatStepper,
+    backwards?: boolean
+  ): void {
+    let seconds = backwards ? currentSeconds : startFrom - currentSeconds;
+    timeline.forEach((element, index) => {
+      if (seconds == element.seconds) {
+        element.isVisible = true;
+        index + this.offset > timeline.length ? timeline[index + this.offset].isVisible = true : null;
+          setTimeout(() => {
+            index - this.offset >= 0 ? timeline[index - this.offset].isVisible = false : null;
+            stepper.selectedIndex = this.offset;
+            this.cdref.detectChanges();
+          }, 90);
+          timeline[index + this.offset + this.offset] !== undefined ? timeline[index + this.offset + this.offset].isVisible = true : null;
+      }
+    });
+  }
+
+  private setVisible(element:ITimeline[], index:number): void {
```

Comment:
Większą uwagę przykładaj do nazw zmiennych oraz metod.

`setVisible` - nie mówi komu zmienia widoczność
`element` - brzmi bardzo ogólnie oraz jak coś pojedynczego, gdzie tak na prawdę jest to tablica czegoś

Reply:
Funkcja usunięta, nie używana.